src/website/shared/models.py

```diff
@@ -255,14 +257,29 @@ class Type(models.TextChoices):
 
 ###
 #
-# Nix related models
+# Nixpkgs related models
 #
 ##
 
 
-class NixIssue(models.Model):
+class NixpkgsIssue(models.Model):
     """The Nixpkgs version of a cve."""
 
+    @property
+    def issue_id(self):
+        return f"NIXPKGS-{str(self.created.year)}-{str(self.number).zfill(5)}"
+
+    created = models.DateField(auto_now_add=True)
+    number = models.IntegerField(default=0)
+
+    def save(self, *args, **kwargs):
+        if not self.pk:
+            max_number = NixpkgsIssue.objects.filter(
```

Comment:
either we make `number` an `models.AutoField()` and we lose the incremental ID **per year** but this is more reliable
either we have to make PostgresSQL use a subselect for the query during update because this thing is vulnerable to TOCTOU.

also please add a integrity constraint on unique_together = ("year", "number") so that the invariant is preserved

Reply:
either case, add a TODO for the first item, add the integrity check please

Reply:
Done